Add_node_tocircularLinkedList.cpp

```diff
@@ -0,0 +1,144 @@
+/*  3.  Add a node to a sorted circular linked list  */
+
+#include <stdio.h>
+#include <conio.h>
+#include <stdlib.h>
+
+void swap(int *x, int *y);
+void print_output();
+int compare(int i, int function);
+struct node *Array_to_CLL(int *arr, int len);
+
+struct node *insert_node_loop_method(struct node* head, int N);					
+
+struct node
+{
+	int num;
+	struct node *next;
+};
+
+struct testcases
+{
+	int arr[100];
+	int len;
+	int N;
+	int output[100];
+	int op_len;
+}testDB[6] = {
+    	{ { 1, 2, 4, 5, 6 }, 5, 3, { 1, 2, 3, 4, 5, 6 }, 6 },
+		{ { 1, 2, 3, 4, 5 }, 5, 6, { 1, 2, 3, 4, 5, 6 }, 6 },
+        { { 1, 2, 2, 3, 3, 3, 5, 5, 7 }, 9, 5, {1, 2, 2, 3, 3, 3, 5, 5, 5, 7}, 10 },
+     	{ { 1, 2, 3 }, 3, NULL, {NULL, 1, 2, 3}, 4 },
+    	{ NULL, 1, 5, { 5 }, 2 },
+     	{ NULL, 1, NULL, NULL, 1 }
+};
+
+void main()
+{
+	print_output();
+	_getch();
+}
+
+void print_output()			
+{
+	int i, count = 9, function_select = 1;
+
+	
+	for (i = 0; i < count; i++)
+	{
+		if (compare(i, function_select) == 1)
+			printf("TESTCASE %d Valid\n", i + 1);
+		else
+			printf("TESTCASE %d Invalid\n", i + 1);
+		
+	}
+}
+
+int compare(int i, int function)		
+{
+	int j, count = 2, flag = 1;
+	struct node *head, *temp;
+	temp = Array_to_CLL(testDB[i].arr, testDB[i].len);
+
+	if (function == 1)
+		head = insert_node_loop_method(temp, testDB[i].N);
+	else
+		head = insert_node_Recursion_method(temp, testDB[i].N, temp);
+
+	for (j = 0, temp = head; j < testDB[i].op_len; j++, temp = temp->next)
+	{
+		if (testDB[i].output[j] != temp->num)
+		{
+			flag = 0;
+			break;
+		}
+	}
+	return flag;
+}
+
+
+struct node *insert_node_loop_method(struct node* head, int N)
+{
+	struct node *i = head, *j = (struct node*)malloc(sizeof(struct node)), *temp1, *ans = head;
+	int flag = 0;
+
+	j->num = N;
+	j->next = NULL;
+	if (head == NULL)
+		return j;
+	else
+	{
+		while (1)
+		{
+			if (i->num >= N || (&i->num == &head->num && flag == 1))
+			{
+				swap(&i->num, &j->num);
```

Comment:
Yes sir, it can be done! just like inserting a node in middle...!
For example : As the linked list contains elements like 1 4 5 6 1 in a circular manner.
if we want to insert a node 3 consider the element which is greater than 3 ie..4 and placing the previous node address to this node 3 and this node points to 4 finally remove the link between 1 and 4